syft/messaging/plan.py

```diff
@@ -220,50 +356,42 @@ class Plan(ObjectStorage, torch.nn.Module):
 
     def __init__(
         self,
-        id: Union[str, int] = None,
-        owner: "sy.workers.BaseWorker" = None,
-        name: str = "",
+        name: str = None,
+        procedure: Procedure = None,
+        state: State = None,
+        include_state: bool = False,
+        is_built: bool = False,
+        # Optional kwargs if commands or state are not provided
         state_ids: List[Union[str, int]] = None,
         arg_ids: List[Union[str, int]] = None,
         result_ids: List[Union[str, int]] = None,
         readable_plan: List = None,
         blueprint=None,
-        state=None,
-        include_state: bool = False,
-        is_built: bool = False,
-        verbose: bool = False,
-        *args,
-        **kwargs,
+        state_dict=None,
+        # General kwargs
+        id: Union[str, int] = None,
+        owner: "sy.workers.BaseWorker" = None,
+        tags: List[str] = None,
+        description: str = None,
     ):
+        owner = owner or sy.local_worker
+        AbstractObject.__init__(self, id, owner, tags, description, child=None)
         ObjectStorage.__init__(self)
         torch.nn.Module.__init__(self)
 
         # Plan instance info
-        self.id = sy.ID_PROVIDER.pop() if id is None else id
-        self.name = self.__class__.__name__ if name == "" else name
-        self._owner = sy.local_worker if owner is None else owner
-        self.verbose = verbose
-
-        # Info about the plan stored
-        self.plan = list()
-        self.readable_plan = readable_plan if readable_plan is not None else []
-        self.state = State(self)
+        self.name = name or self.__class__.__name__
+        self.owner = owner
+
+        # Info about the plan stored vua the state and the procedure
+        self.procedure = procedure or Procedure(readable_plan, arg_ids, result_ids)
+        self.state = state or State(owner=owner, plan=self, state_ids=state_ids)
+        if state_dict is not None:
```

Comment:
this also feels pretty torch-specific

Reply:
how so?

Reply:
iirc, `state_dict` is how torch returns stateful objects within a container (e.g. variable tensors from an optimizer or nn.Module)

Reply:
That code is outdated; now is used a tuple